src/test/scala-2.13/ru/delimobil/cabbit/ConfigSuite.scala

```diff
@@ -0,0 +1,145 @@
+package ru.delimobil.cabbit
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import com.rabbitmq.client.BuiltinExchangeType
+import com.typesafe.config.ConfigFactory.parseString
+import com.typesafe.config.ConfigValue
+import org.scalatest.funsuite.AnyFunSuite
+import pureconfig.ConfigReader
+import pureconfig.ConfigSource
+import pureconfig.generic.auto._
+import pureconfig.module.cats._
+import ru.delimobil.cabbit.algebra.ExchangeName
+import ru.delimobil.cabbit.algebra.QueueName
+import ru.delimobil.cabbit.config.CabbitConfig
+import ru.delimobil.cabbit.config.CabbitConfig.CabbitNodeConfig
+import ru.delimobil.cabbit.config.declaration._
+
+import scala.concurrent.duration._
+
+class ConfigSuite extends AnyFunSuite {
+
+  implicit val reader: ConfigReader[Map[String, Any]] =
+    ConfigReader[Map[String, ConfigValue]].map(_.map { case(s, value) => s -> value.unwrapped() })
+
+  test("CabbitConfig can be extracted by pureconfig") {
+    val conf = parseString("""nodes = [
+                             |  {
+                             |    host: "localhost",
+                             |    port: 5672
+                             |  }
+                             |]
+                             |virtual-host = "/"
+                             |connection-timeout = 60.seconds
+                             |""".stripMargin)
+
+    val expected =
+      CabbitConfig(
+        nodes = NonEmptyList(CabbitNodeConfig("localhost", 5672), List.empty),
```

Comment:
Requires pureconfig-cats, which hasn't yet been ported to dotty. Thus, is tested for scala 2.13 only.